components/kyma-metrics-collector/pkg/process/process.go

```diff
@@ -399,29 +406,45 @@ func (p *Process) populateCacheAndQueue(runtimes *kebruntime.RuntimesPage) {
 					// The shootname has changed hence the record in the cache is not valid anymore
 					// No need to queue as the subAccountID already exists in queue
 					p.Cache.Set(runtime.SubAccountID, newRecord, cache.NoExpiration)
-					p.namedLogger().With(log.KeySubAccountID, runtime.SubAccountID).Debug("Resetted the values in cache for subAccount")
+					p.namedLogger().With(log.KeySubAccountID, runtime.SubAccountID).With(log.KeyRuntimeID, runtime.RuntimeID).
+						Debug("Resetted the values in cache for subAccount")
 				}
 			}
 			continue
 		}
 		if isFoundInCache {
 			// Cluster is not trackable but is found in cache should be deleted
 			p.Cache.Delete(runtime.SubAccountID)
-			p.namedLogger().With(log.KeySubAccountID, runtime.SubAccountID).Debug("Deleted subAccount from cache")
+			p.namedLogger().With(log.KeySubAccountID, runtime.SubAccountID).
+				With(log.KeyRuntimeID, runtime.RuntimeID).Debug("Deleted subAccount from cache")
 			continue
 		}
-		p.namedLogger().With(log.KeySubAccountID, runtime.SubAccountID).Debug("Ignoring SubAccount as it is not trackable")
+		p.namedLogger().With(log.KeySubAccountID, runtime.SubAccountID).
+			With(log.KeyRuntimeID, runtime.RuntimeID).Debug("Ignoring SubAccount as it is not trackable")
 	}
 
 	// Cleaning up subAccounts from the cache which are not returned by KEB anymore
-	for sAccID := range p.Cache.Items() {
+	for sAccID, recordObj := range p.Cache.Items() {
 		if _, ok := validSubAccounts[sAccID]; !ok {
+			record, ok := recordObj.Object.(kmccache.Record)
+			if !ok {
+				p.namedLogger().With(log.KeySubAccountID, sAccID).
+					Error("bad item from cache, could not cast to a record obj")
```

Comment:
Should we add a `continue` statement here?

```golang
if !ok {
  // stuff
  continue
}
```

Reply:
resolved